test/lib/requests.go

```diff
@@ -0,0 +1,272 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lib
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/test/monitoring"
+
+	"knative.dev/pkg/test/logging"
+)
+
+// Port for the recordevents pod REST listener
+const RecordEventsPort = 8392
+
+// HTTP path for the GetMinMax REST call
+const GetMinMaxPath = "/minmax"
+
+// HTTP path for the GetEntry REST call
+const GetEntryPath = "/entry/"
+
+// HTTP path for the TrimThrough REST call
+const TrimThroughPath = "/trimthrough/"
+
+// On-wire json rest api format for recordevents GetMinMax calls
+// sennt to the recordevents pod.
+type MinMaxResponse struct {
+	MinAvail int
+	MaxSeen  int
+}
+
+// Structure to hold information about an event seen by recordevents pod.
+type EventInfo struct {
+	// Set if the cloudevent received by the pod didn't pass validation
+	ValidationError string
+	// Event received if the cloudevent received by the pod passed validation
+	Event *cloudevents.Event
+	// HTTPHeaders of the connection that delivered the event
+	HTTPHeaders map[string][]string
+}
+
+// Pretty print the event. Meant for debugging.  This formats the validation error
+// or the full event as appropriate.  This does NOT format the headers.
+func (ei *EventInfo) String() string {
+	if ei.Event != nil {
+		return ei.Event.String()
+	} else {
+		return fmt.Sprintf("invalid event \"%s\"", ei.ValidationError)
+	}
+}
+
+// This is mainly used for providing better failure messages
+type SearchedInfo struct {
+	TotalEvent int
+	LastNEvent []EventInfo
+}
+
+// Pretty print the SearchedInfor for error messages
+func (s *SearchedInfo) String() string {
+	return fmt.Sprintf("%d events seen, last N = %s", s.TotalEvent, s.LastNEvent)
+}
+
+// Connection state for a REST connection to a pod
+type eventGetter struct {
+	podName       string
+	podNamespace  string
+	podPort       int
+	kubeClientset *kubernetes.Clientset
+	logf          logging.FormatLogger
+
+	host       string
+	port       int
+	forwardPID int
+}
+
+// Creates a forwarded port to the specified recordevents pod and waits until
+// it can successfully talk to the REST API.  Times out after 4 minutes
```

Comment:
I would rather point to the timeout variable that we are using... so that if we change the timeout to be != 4 mins later on, we don't have to change all these comments... also, a nit and optional

Reply:
fixed